com.unity.shadergraph/Editor/Drawing/Views/GraphSubWindow.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEditor.Experimental.GraphView;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEngine;
+using UnityEngine.UIElements;
+using ResizableElement = UnityEditor.ShaderGraph.Drawing.ResizableElement;
+
+namespace Drawing.Views
+{
+    public class GraphSubWindow : GraphElement, ISelection
+    {
+        protected VisualElement m_MainContainer;
+        protected VisualElement m_Root;
+        protected Label m_TitleLabel;
+        protected Label m_SubTitleLabel;
+        protected ScrollView m_ScrollView;
+        protected VisualElement m_ContentContainer;
+        protected VisualElement m_HeaderItem;
+
+        private bool m_Scrollable = false;
+
+        private Dragger m_Dragger;
+        protected GraphView m_GraphView;
+
+        WindowDockingLayout m_Layout;
+        WindowDockingLayout m_DefaultLayout = new WindowDockingLayout
+        {
+            dockingTop = true,
+            dockingLeft = true,
+            verticalOffset = 16,
+            horizontalOffset = 16,
+            size = new Vector2(200, 400),
+        };
+
+        private const string UxmlName = "GraphSubWindow";
+
+        // Each sub-window will override these if they need to
+        protected virtual string windowTitle => "SubWindow";
```

Comment:
Do we ever want to show these actual strings? I'd say no. I think they should default to "".